src/store.ts

```diff
@@ -413,32 +409,115 @@ export default function Store(
                 this.SDKConfig.onCreateBatch = undefined;
             }
         }
+    }
+}
 
-        if (!config.hasOwnProperty('flags')) {
-            this.SDKConfig.flags = {};
-        }
-        if (
-            !this.SDKConfig.flags.hasOwnProperty(
-                Constants.FeatureFlags.EventBatchingIntervalMillis
-            )
-        ) {
-            this.SDKConfig.flags[
-                Constants.FeatureFlags.EventBatchingIntervalMillis
-            ] = Constants.DefaultConfig.uploadInterval;
-        }
-        if (
-            !this.SDKConfig.flags.hasOwnProperty(
-                Constants.FeatureFlags.ReportBatching
-            )
-        ) {
-            this.SDKConfig.flags[Constants.FeatureFlags.ReportBatching] = false;
-        }
-        if (
-            !this.SDKConfig.flags.hasOwnProperty(
-                Constants.FeatureFlags.OfflineStorage
-            )
-        ) {
-            this.SDKConfig.flags[Constants.FeatureFlags.OfflineStorage] = 0;
+export interface IFeatureFlags {
```

Comment:
Can you please move this to the top of the file? I think all interfaces should live up there.